### ISSUE TITLE
[UPSTREAM] Touch keyboard does not open on windows

### ISSUE THREAD

**Issue by schneidr** [telegramdesktop/tdesktop#218]
The Windows onscreen Keyboard does not open automatically when focusing the Input field on a device without physical Keyboard (tablet).

tdesktop Version: 0.6.5

**Comment by Alex131089** [comment]
+1, still not happening.
Dev is looking for some documentations on how to do this with Qt, if anyone have information & can help.

**Comment by Alex131089** [comment]
@telegramdesktop : while being ugly, this seems an adopted solution : https://github.com/qreal/qreal/blob/e7d2aeeb6c38fffef3686fe85b73e25772885e63/qrutils/virtualKeyboard.cpp (call the binary directly)

**Comment by Alex131089** [comment]
More informations : 
- https://forum.qt.io/topic/29192/windows-8-keyboard/2
- https://code.launchpad.net/~baratosha23/stellarium/stellarium/+merge/237364
- http://forum.qt.io/topic/5405/textinput-and-opensoftwareinputpanel-on-windows/4
- http://doc.qt.io/QtVirtualKeyboard/qtvirtualkeyboard-enterprise-virtualkeyboard-virtualkeyboard-virtualkeyboard-qml.html
(and http://blogs.msdn.com/b/b8/archive/2012/07/17/designing-the-windows-8-touch-keyboard.aspx if anyone interested)

**Comment by Alex131089** [comment]
Another example : https://chromium.googlesource.com/chromium/src/+/lkgr/base/win/win_util.cc#393

**Comment by Snnlk** [comment]
I can confirm this is still a problem in Windows 10 (in tablet mode).

**Comment by RPDeshaies** [comment]
:+1: I'm in december and the problem is still happening.
Is there plan to fix this issue ? I cant use the app on my W10 tablet because of that

**Comment by turnitepp** [comment]
this is STILL and issue.

**Comment by 4ndv** [comment]
Still not fixed

**Comment by telegramdesktop** [comment]
@4ndv This is a Qt issue: https://bugreports.qt.io/browse/QTBUG-43190 :(

I'm afraid I don't understand Qt well enough to fix this yet.

**Comment by Alex131089** [comment]
Can't you add an event when the text zone is active ? If so, the workaround is just to call tabtip.exe as in https://chromium.googlesource.com/chromium/src/+/lkgr/base/win/win_util.cc#448, on the event.

**Comment by alecuba16** [comment]
Same problem, there is some work around?

**Comment by yasharne** [comment]
same here

**Comment by RussianNeuroMancer** [comment]
Same issue in Linux with Onboard.

**Comment by grkashani** [comment]
same here , why s not fixed yet

**Comment by RussianNeuroMancer** [comment]
Reproducible not only with Onboard, but also with Caribou too (Gnome Shell integrated keyboard).

**Comment by alskrypnik** [comment]
It seem's like known QT bug assotiated with this issue has been fixed: https://bugreports.qt.io/browse/QTBUG-43190

**Comment by MarcAnt01** [comment]
lol, 2019 and still not fixed😂😂

**Comment by mirh** [comment]
Because qt fixed the bug more or less one year ago, but telegram's still using 5.6.2. 
So.. perhaps it depends on #1815

**Comment by MarcAnt01** [comment]
> Because qt fixed the bug more or less one year ago, but telegram's still using 5.6.2.
> So.. perhaps it depends on #1815

Thanks for giving a detailed answer, I hope it'll be updated

**Comment by RussianNeuroMancer** [comment]
@mirh 

> Because qt fixed the bug more or less one year ago, but telegram's still using 5.6.2.

Do you sure this is related to Qt version? I tested [Telegram build with Qt 5.12.2](https://packages.ubuntu.com/disco/telegram-desktop) (check libqt5core5a version in dependency list) and issue is sill there.

**Comment by mirh** [comment]
I thought this bug (and the linked tickets) just covered Windows?

**Comment by RussianNeuroMancer** [comment]
Indeed, I register [separate issue](https://github.com/telegramdesktop/tdesktop/issues/6402).

**Comment by MarcAnt01** [comment]
> I thought this bug (and the linked tickets) just covered Windows?

Yes, but the functionality seems the same across OSs

**Comment by Aokromes** [comment]
maybe this can help?
https://forum.qt.io/topic/84518/show-keyboard-when-qtextedit-is-activated/9

**Comment by ilya-fedin** [comment]
> Same issue in Linux with Onboard.

Does it work with any other Qt app (VLC, qBittorrent)?

**Comment by pitlochry** [comment]
@stale. After more than six years this is still an issue on windows.

**Comment by MarcAnt01** [comment]
No doubt, they don't realize that they sell tablets with Windows 10, but no problem: I totally uninstalled Telegram Desktop now.

**Comment by Aokromes** [comment]
think it's only one dev doing most of work.

**Comment by ilya-fedin** [comment]
I was thinking this is fixed for Windows users since 1.9.0 due to the Qt update

**Comment by RussianNeuroMancer** [comment]
> due to the Qt update

What version of Qt is required? Yesterday issue was still reproducible with Telegram Desktop 2.2.0 and Qt 5.14.2 on Ubuntu 20.10.

Or Linux version of Qt is still missing on-sceen keyboard support?

**Comment by MarcAnt01** [comment]
> think it's only one dev doing most of work.

I know, but in 2020 every app should support touch keyboard and I don't know another app apart from Telegram Desktop that doesn't do that. It's more than 5 years and no fix has been made for tablet users forcing us to move to something else.

**Comment by ilya-fedin** [comment]
> What version of Qt is required?

> I was thinking this is fixed **for Windows** users since 1.9.0 due to the Qt update

**Comment by ilya-fedin** [comment]
> Or Linux version of Qt is still missing on-sceen keyboard support?

I don't know, you need to contact Qt developers

**Comment by ilya-fedin** [comment]
> I know, but in 2020 every app should support touch keyboard

Sorry, but this is something that toolkit should care of, not app developers

**Comment by RussianNeuroMancer** [comment]
> you need to contact Qt developers

I assume regular users is not supposed to do this? I mean, Qt bugtracker is supposed to be used by people who use Qt - app developers who use Qt for development, know C++, can understand toolkit documentation, etc. (I can't do any of this.)

**Comment by ilya-fedin** [comment]
> I assume regular users is not supposed to do this?

Not in case of Linux, Linux users usually creating reports in Qt bugtracker on their own.

**Comment by MarcAnt01** [comment]
> > I know, but in 2020 every app should support touch keyboard
> 
> Sorry, but this is something that toolkit should care of, not app developers

If there is no fix QT side, another fix should be made. This is a critical bug and what I see here is just justifying it.

**Comment by ilya-fedin** [comment]
This doesn't look like a critical bug, IMHO

**Comment by ilya-fedin** [comment]
> If there is no fix QT side, another fix should be made.

I disagree, hijacking things that should be done on toolkit level would lead into bad consequences

**Comment by MarcAnt01** [comment]
> This doesn't look like a critical bug, IMHO

Because you don't own a tablet, try to buy a Windows tablet and use it without a keyboard, then you'll understand

**Comment by ilya-fedin** [comment]
This still would be a bug related only for touch users

**Comment by MarcAnt01** [comment]
So, if Telegram desktop doesn't work with mouse it's only a problem for mouse users.

**Comment by ilya-fedin** [comment]
Yeah, sounds like true

**Comment by RussianNeuroMancer** [comment]
> Not in case of Linux, Linux users usually creating reports in Qt bugtracker on their own.

Then I should try latest Qt version to make bugreport actually useful. But I can't, because latest version available in Ubuntu is 5.14.2, while latest Qt version 5.15.1. I have no idea how to do so because I am not developer.

And if would be much simpler for developer to register bugreport about this toolkit issue. Is there specific reason why you wouldn't like to communicate this issue issue to toolkit developers to make your app usable for users who want to use it, but can't because of this issue?

**Comment by ilya-fedin** [comment]
> I have no idea how to do so because I am not developer.

Install a rolling release distro

**Comment by RussianNeuroMancer** [comment]
@ilya-fedin how to do so in case of convertible with Snapdragon SoC? (Lenovo Yoga C630 WOS) All available [information](https://github.com/aarch64-laptops/build), [pre-build images](http://releases.linaro.org/aarch64-laptops/images/), [kernels](https://yadi.sk/d/FiLcBs2hszq_Jg/Lenovo%20Yoga%20C630%20WOS), [additional system services](https://yadi.sk/d/FiLcBs2hszq_Jg) (qmi-svc) is for Ubuntu, as you can see. There is also [manual](https://steev.net/content/post/convert-c630-bionic-kali/) on how to convert Ubuntu installation into Kali Linux which is rolling release, as you suggested, but [Qt version is 5.14.2 here](https://pkg.kali.org/pkg/qtbase-opensource-src) too.

Of course I have access to a few other devices with touchscreens with x86 CPU from time to time, such as [HP Elite x2 1013 G1](https://github.com/systemd/systemd/pull/17174) and [Irbis NB111](https://bugzilla.kernel.org/show_bug.cgi?id=209397) - but these hardware isn't my own. I get it on hands to solve specific issue and then I return it to the owners (and some of them complain about this issue with Telegram, btw). Changing operating system on other people devices is not an option for me, as you hopefully understand, especially in a limited time frame I have hardware on hands.

Could you please clarify, is there specific reason why you wouldn't like to communicate this issue issue to toolkit developers to make your app usable for users who want to use it, but can't because of this issue?

**Comment by laozhoubuluo** [comment]
**This is still an issue on Tablet PC after more than six years.**

**Comment by ilya-fedin** [comment]
> This is still an issue on Tablet PC after more than six years.

Code is open source, you can fix it yourself if you know how to.

**Comment by MarcAnt01** [comment]
> **This is still an issue on Tablet PC after more than six years.**

If you have Windows 10 try Unigram, I don't think this bug will be ever fixed

**Comment by monte-monte** [comment]
Ubuntu 20.10, Gnome 3.38.2
Telegram desktop is unusable without keyboard. Please fix this critical bug.

**Comment by ilya-fedin** [comment]
> Telegram desktop is unusable without keyboard. Please fix this critical bug.

Please provide a way to fix it

**Comment by MarcAnt01** [comment]
> > Telegram desktop is unusable without keyboard. Please fix this critical bug.
> 
> Please provide a way to fix it

As users we aren't supposed to

**Comment by monte-monte** [comment]
@ilya-fedin if I had a way to fix it in mind, I wouldn't be on this page talking to you.

**Comment by MarcAnt01** [comment]
The thing is that we are not in an open source project by a random dude who provides their work for free. This is a company project and as a final user I don't accept a critical bug like this one to have been ignored and people justifying them just because they are not affected.
As for me I am happy using another client, but people on Linux will still have this problem and no workaround

**Comment by ilya-fedin** [comment]
Personally I'm an third-party contributor. The only who hired is @john-preston and he sadly doesn't know Linux APIs at all (just remember the old appindicator tray)

**Comment by ilya-fedin** [comment]
> if I had a way to fix it in mind, I wouldn't be on this page talking to you.

Probably this never will be fixed... because no one knows how to implement this

**Comment by laozhoubuluo** [comment]
> 
> 
> Hey there!
> 
> This issue was inactive for a long time and will be automatically closed in 30 days if there isn't any further activity. We therefore assume that the user has lost interest or resolved the problem on their own.
> 
> Don't worry though; if this is an error, let us know with a comment and we'll be happy to reopen the issue.
> 
> Thanks!

Probably this never will be fixed, but this issue should be saved and keep it open.

**Comment by ilya-fedin** [comment]
Is something changed in 3.2.x? While porting to Qt 6 it was found that Qt asks the system to open keyboard in mouseReleaseEvent while Telegram sets WA_AcceptTouchEvent on its widgets and gets TouchEnd events instead.

**Comment by Aokromes** [comment]
on windows remember enabling option one
https://www.tenforums.com/tutorials/83312-turn-off-automatically-show-touch-keyboard-windows-10-a.html
it will work.
